File: java/src/test/java/org/katas/CurrencyConverterTest.java
package org.katas;

import org.katas.external.ConversionRateApi;
import org.junit.jupiter.api.Test;
import org.katas.model.Currency;
import org.katas.model.Money;

import static org.junit.jupiter.api.Assertions.assertEquals;


class CurrencyConverterTest {

    @Test
    void shouldWork() {
        assertEquals(3, 1+2);
    }

    @Test
    void sum_shouldFail() {
        var converter = new CurrencyConverter(new ConversionRateApi());

        // This does not work. Make it work please ^_^"
        var result = converter.sum(Currency.Euro, new Money(1, Currency.Dollar));
    }
}

File: kotlin/src/test/kotlin/org/katas/CurrencyConverterTest.kt
package org.katas

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.katas.external.conversionRateApi.ConversionRateApi
import org.katas.model.Currency
import org.katas.model.Money

class CurrencyConverterTest {
    @Test
    fun shouldWork() {
        Assertions.assertEquals(3, 1 + 2)
    }

    @Test
    fun sum_shouldFail() {
        val converter = CurrencyConverter(ConversionRateApi())

        // This does not work. Make it work please ^_^"
        val result = converter.sum(Currency.EURO, Money(1.0, Currency.DOLLAR))
    }
}